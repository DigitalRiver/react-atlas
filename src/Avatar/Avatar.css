.avatar {
  composes: circle from '../styles.css';
  composes: white from '../styles.css';
  composes: bg-silver from '../styles.css';
  composes: default-border from '../styles.css';
  composes: default-font from '../styles.css';
  position: relative;
  display: inline-block;
  overflow: hidden;
  text-align: center;
  width: 3rem;
  height: 3rem;
  font-size: 1.9rem;
}

.avatar > :global(svg),
.avatar > :global(i) {
  width: 2rem;
  height: 3rem;
  line-height: 3rem;
  fill: currentColor;
}

.avatar > :global(img) {
  width: 100%;
  height: 100%;
}

.image {
  composes: circle from '../styles.css';
  composes: bg-transparent from '../styles.css';
  position: absolute;
  display: block;
  width: 100%;
  height: 100%;
  background-position: center;
  background-size: cover;
}

.letter {
  display: block;
  width: 100%;
  line-height: 3rem;
  cursor: default;
}
