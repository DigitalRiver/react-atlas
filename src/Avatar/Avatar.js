import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Avatar.css";
import blacklist from "blacklist";

/**
 * Avatar component creates a circular area where an image, letter or icon/glyphicon can be presented. Great for user profiles and lists.
 * **NOTE**: children will always take precedence over props passed into component.
 * Precedence Order: image, fallbackImage, icon, title.
 */
export class Avatar extends React.PureComponent {
  constructor(props) {
    super(props);
    let image = props.image ? props.image : props.fallbackImage;
    this.state = { image: image ? image : null };
  }

  componentDidUpdate = prevProps => {
    if (this.props.image !== prevProps.image) {
      this.setState({ image: this.props.image });
    }
  };

  _handleBadImage = () => {
    /* If the default Image is equal to the bad image URL or the default image is undefined
        set this.state.image as null so avatar will fallback on a different prop. */
    if (
      this.props.fallbackImage &&
      this.props.fallbackImage !== this.state.image
    ) {
      this.setState({ image: this.props.fallbackImage });
      return;
    }
    this.setState({ image: null });
  };

  _getContent = () => {
    if (this.props.children) {
      return this.props.children;
    }
    let avatar = null;
    if (this.state.image) {
      avatar = (
        <img
          src={this.state.image}
          title={this.props.title}
          onError={this._handleBadImage}
          styleName={"image"}
        />
      );
    } else if (this.props.icon) {
      avatar = <i className={this.props.icon} />;
    } else if (this.props.title) {
      avatar = <span styleName={"letter"}>{this.props.title[0]}</span>;
    }
    return avatar;
  };

  render() {
    let { title, className, style, ...others } = this.props;

    // Declaring the following variables so they don't get passed to TextField through the prop spread.
    const othersFiltered = blacklist(
      others,
      "children",
      "fallbackImage",
      "icon",
      "image"
    );

    return (
      <div
        {...othersFiltered}
        title={title}
        className={cx(className)}
        styleName={"avatar"}
        style={style}
      >
        {this._getContent()}
      </div>
    );
  }
}

Avatar.propTypes = {
  /**
   * Children should be either a string, an icon/glyphicon, or an image tag.
   * @examples "SomeName", <SomeIcon />, <img src="/path/to/image.jpg"/>
   */
  children: PropTypes.node,
  /** An object, array, or string of CSS classes to apply to Avatar.*/
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * A URL for an image that will be displayed when the main image fails to load.
   */
  fallbackImage: PropTypes.string,
  /**
   * For displaying an icon/glphyicon. Usually another component or an element with a class on it.
   * @examples <GithubIcon />, <i class="fa fa-github"></i>
   */
  icon: PropTypes.string,
  /**
   * Path to an image.
   * @examples "http://path.to/an/image.jpg"
   */
  image: PropTypes.string,
  /**
   * Pass inline styles here.
   */
  style: PropTypes.object,
  /**
   * Avatar will use the first letter of the title string when children, image, and icon are not supplied.
   * @examples "Nathan" will output "N"
   */
  title: PropTypes.string
};

export default CSSModules(Avatar, styles, { allowMultiple: true });
