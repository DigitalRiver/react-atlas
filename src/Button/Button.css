@import '../styles/variables.css';

/*Main Styles*/
.base {
  composes: button-marg-1  from '../styles.css';
  composes: button-pad-1 from '../styles.css';
  composes: default-text from '../styles.css';
  composes: cursor-pointer from '../styles.css';
  composes: primary-button-border-width from '../styles.css';
  composes: default-font from '../styles.css';
  composes: rounded from '../styles.css';
  text-decoration: none;
  display: inline-block;
  vertical-align: middle;
  line-height: 1.125rem;
  -webkit-appearance: none;
  font-size: 1.0625rem;
  border-style: solid;
}

.uppercase {
  text-transform: uppercase;
}

.base:hover {
  text-decoration: none;
}

.default_btn {
  composes: base;
  composes: bg-default from '../styles.css';
  composes: white from '../styles.css';
  border: var(--border-width) solid var(--default);

}
.default_btn_hover {
  composes: hover-bg-gray from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-default from '../styles.css';
}
.default_btn:active {
  box-shadow: inset 0 0 0 20rem var(--darken-2),
  inset 0 3px 4px 0 var(--darken-3),
  0 0 1px var(--darken-2);
}
.default_btn:disabled {
  box-shadow: inset 0 0 0 20rem var(--darken-1);
}

.outline {
  composes: base;
  composes: bg-transparent from '../styles.css';
}
.outline:active {
  box-shadow: inset 0 0 0 20rem var(--darken-2),
  inset 0 3px 4px 0 var(--darken-3),
  0 0 1px var(--darken-2);
}
.outline:disabled {
  opacity: .5;
}

/*default button variant*/
.primary {
  composes: bg-primary from '../styles.css';
  composes: border-primary from '../styles.css';
  composes: white from '../styles.css';
}
.primary_hover {
  composes: hover-bg-primary from '../styles.css';
  composes: hover-border-primary from '../styles.css';
}
.button {
  composes: base;
}
.secondary {
  composes: bg-secondary from '../styles.css';
  composes: border-secondary from '../styles.css';
  composes: inverse-blue-text from '../styles.css';
}
.secondary_hover {
  composes: hover-bg-secondary from '../styles.css';
  composes: hover-border-secondary from '../styles.css';
}

.warning {
  composes: white from '../styles.css';
  composes: bg-warning from '../styles.css';
  composes: border-warning from '../styles.css';
}
.warning_hover {
  composes: hover-bg-warning from '../styles.css';
  composes: hover-border-warning from '../styles.css';
}

/* TODO: Remove hardcoded color values and
 * compose from styles.css */
.error {
  composes: white from '../styles.css';
  composes: bg-error from '../styles.css';
  composes: border-error from '../styles.css';
}
.error_hover {
  composes: hover-bg-error from '../styles.css';
  composes: hover-border-error from '../styles.css';
}

.link {
  composes: base;
  composes: border-none from '../styles.css';
  composes: sky-blue from '../styles.css';
  composes: border-transparent from '../styles.css';
  composes: bg-transparent from '../styles.css';
}
.link_hover {
  composes: hover-bg-link from '../styles.css';
  composes: hover-secondary-text from '../styles.css';
}

/*Outline styles*/
.default_btn_outline {
  composes:outline;
  composes: hover-bg-gray from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-default from '../styles.css';
}
.primary_outline{
  composes: outline;
  composes: sky-blue from '../styles.css';
  composes: hover-bg-primary from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-primary from '../styles.css';
  composes: hover-border-primary from '../styles.css';
}
.button_outline {
  composes: outline;
  composes: hover-bg-gray from '../styles.css';
  composes: hover-border-default from '../styles.css';
  composes: hover-white from '../styles.css';
}
.secondary_outline {
  composes: outline;
  composes: blue from '../styles.css';
  composes: hover-bg-secondary from '../styles.css';
  composes: border-secondary from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: hover-border-secondary from '../styles.css';
}

.success_outline {
  composes: outline;
  composes: green from '../styles.css';
  composes: hover-bg-green from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-green from '../styles.css';
}

.warning_outline {
  composes: outline;
  composes: orange from '../styles.css';
  composes: hover-bg-warning from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-warning from '../styles.css';
  composes: hover-border-warning from '../styles.css';
}

.error_outline {
  composes: outline;
  composes: red from '../styles.css';
  composes: hover-bg-error from '../styles.css';
  composes: hover-white from '../styles.css';
  composes: border-error from '../styles.css';
  composes: hover-border-error from '../styles.css';
}

.link_outline {
  composes: outline;
  composes: primary-text from '../styles.css';
  composes: bg-transparent from '../styles.css';
  composes: border-none from '../styles.css';
}
.link_outline:hover {
  text-decoration:underline;
}
.disabled {
  composes: cursor-not-allowed from '../styles.css';
  text-decoration: none;
  opacity: .65;
}

/*button sizes*/
.large {
  padding: 1rem 1.25rem;
  font-size: 1.25rem;
}

.small {
  padding: .25rem .5rem;
  font-size: .85rem;
}

/*block style*/
.block {
  display: block;
  width: 100%;
}

.icon-left {
  margin-right: 1rem;
}