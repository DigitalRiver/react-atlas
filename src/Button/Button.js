import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Button.css";

/**
 * A Generic button component.
 */
export class Button extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onClick,
      className,
      children,
      disabled,
      large,
      small,
      primary,
      secondary,
      warning,
      error,
      link,
      outline,
      icon,
      type,
      href,
      ignoreTab,
      id,
      name,
      style
    } = this.props;

    let mainStyle = "button";
    let hoverStyle = "";
    if (primary) {
      mainStyle = "primary";
      if (!disabled) {
        hoverStyle = "primary_hover";
      }
    } else if (secondary) {
      mainStyle = "secondary";
      if (!disabled) {
        hoverStyle = "secondary_hover";
      }
    } else if (warning) {
      mainStyle = "warning";
      if (!disabled) {
        hoverStyle = "warning_hover";
      }
    } else if (error) {
      mainStyle = "error";
      if (!disabled) {
        hoverStyle = "error_hover";
      }
    } else if (link) {
      mainStyle = "link";
      if (!disabled) {
        hoverStyle = "link_hover";
      }
    } else {
      mainStyle = "default_btn";
      if (!disabled) {
        hoverStyle = "default_btn_hover";
      }
    }

    if (outline) {
      mainStyle += "_outline";
    }

    /**
     * Use cx to construct a CSS class object.
     * cx will automatically remove falsy values
     * for us. This way only passed props will be
     * transformed into CSS classes.
     */
    const classes = cx(
      {
        "disabled": disabled,
        "large": large,
        "small": small,
        "button": true,
        "uppercase": !link
      },
      mainStyle,
      hoverStyle
    );

    let text = children;
    let iconClass = false;

    /* Just use the icon instead of text when an icon
     * is passed and no children(text) is set. */
    if (icon && children === "Default Button") {
      text = null;
    } else if (icon) {
      text = children;
      iconClass = "icon-left";
    }

    let btn = 
      <button
        onClick={disabled ? null : onClick}
        className={cx(className)}
        styleName={classes}
        style={style}
        type={disabled ? Button.defaultProps.type : type}
        href={href}
        tabIndex={ignoreTab ? -1 : 0}
        id={id}
        name={name}
      >
        {icon ? <i className={icon} styleName={iconClass} /> : null}
        {text}
      </button>
    ;

    let renderButton = btn;
    if (href) {
      renderButton = <a href={href}>{btn}</a>;
    }

    return renderButton;
  }
}

Button.propTypes = {
  /**
   * Children should be either a string, an icon/glyphicon, or an image tag.
   *
   */
  "children": PropTypes.node,
  /** An object, array, or string of CSS classes to apply to Button.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   *When true, Button component will be disabled.
   * @examples <Button disabled/>
   */
  "disabled": PropTypes.bool,
  /**
   * Will display error style button.
   * @examples <Button error>
   */
  "error": PropTypes.bool,
  /** The URL that the Button will link to. */
  "href": PropTypes.string,
  /**
   * The className of the icon you want to set.
   */
  "icon": PropTypes.string,
  /**
   * Will set the html "id" property on the Button.
   */
  "id": PropTypes.string,
  /**
   * Button will be ignored by keyboard navigation when set to true.
   */
  "ignoreTab": PropTypes.bool,
  /**
   * Will display large style Button.
   * @examples <Button large>
   */
  "large": PropTypes.bool,
  /**
   * Will display large style Button.
   * @examples <Button link>
   */
  "link": PropTypes.bool,
  /**
   * Will set the html "name" property on the Button.
   */
  "name": PropTypes.string,
  /**
   * Click event handler for the Button.
   */
  "onClick": PropTypes.func,
  /**
   * Will display outline styled Button.
   * @examples <Button outline>
   */
  "outline": PropTypes.bool,
  /**
   * Will display primary style Button.
   * @examples <Button primary>
   */
  "primary": PropTypes.bool,
  /**
   * Will display secondary style Button.
   * @examples <Button secondary>
   */
  "secondary": PropTypes.bool,
  /**
   * Will display small style Button.
   * @examples <Button small>
   */
  "small": PropTypes.bool,
  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,
  /** HTML5 button type, eg submit, button, etc. */
  "type": PropTypes.string,
  /**
   * Will display warning style Button.
   * @examples <Button warning>
   */
  "warning": PropTypes.bool
};

Button.defaultProps = {
  "children": "Default Button",
  "outline": false,
  "primary": false,
  "secondary": false,
  "warning": false,
  "error": false,
  "link": false,
  "large": false,
  "small": false,
  "disabled": false,
  "type": "button",
  "ignoreTab": false
};
export default CSSModules(Button, styles, { "allowMultiple": true });
