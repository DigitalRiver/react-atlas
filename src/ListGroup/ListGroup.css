.divider {
	composes: border-bottom from '../styles.css';
}

.groupTitle {
	composes: pad-v-1 from '../styles.css';
	composes: header-font from '../styles.css';
}