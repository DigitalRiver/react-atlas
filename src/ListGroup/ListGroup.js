import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./ListGroup.css";

export class ListGroup extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { children, className, divider, id, style, title } = this.props;

    const groupStyles = cx({
      divider
    });

    return (
      <div id={id} style={style} className={className} styleName={groupStyles}>
        <div styleName="groupTitle">{title}</div>
        {children}
      </div>
    );
  }
}

ListGroup.propTypes = {
  /**
   * Text, any HTML element, or React Component.
   */
  "children": PropTypes.node,
  /** An object, array, or string of CSS classes to apply to Input. */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /** When true, will add a horizontal rule between the ListGroup and subsequent children. */
  "divider": PropTypes.bool,
  /**
   * Will set the html "id" property for ListGroup.
   */
  "id": PropTypes.string,
  /** Pass inline styling here. */
  "style": PropTypes.object,
  /** Will add a title to the ListGroup. */
  "title": PropTypes.string
};

ListGroup.defaultProps = {
  "divider": true
};

export default CSSModules(ListGroup, styles, { "allowMultiple": true });
