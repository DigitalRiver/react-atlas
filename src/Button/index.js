import Button from "./Button.js";
export { Button };
