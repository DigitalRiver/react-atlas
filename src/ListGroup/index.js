import ListGroup from "./ListGroup.js";
export { ListGroup };
