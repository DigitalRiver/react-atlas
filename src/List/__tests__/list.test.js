import React from "react";
import { List } from "../index";
import { ListGroup } from "../../ListGroup/index";
import { ListItem } from "../../ListItem/index";
import { Avatar } from "../../Avatar/index";
import { Switch } from "../../Switch/index";
import { Button } from "../../Button/index";
import renderer from "react-test-renderer";

describe("Test List component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <List border>
          <ListGroup title="Dinosaurs">
            <ListItem leftItem={<Avatar title="Blue" />} rightItem={<Switch />}>
              <Button link>Blue</Button>
            </ListItem>
            <ListItem leftItem={<Avatar title="Earl" />} rightItem={<Switch />}>
              <Button link>Earl Sinclair</Button>
            </ListItem>
          </ListGroup>
        </List>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
