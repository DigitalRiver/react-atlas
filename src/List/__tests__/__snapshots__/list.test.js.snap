// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List component renders correctly 1`] = `
<div
  className="list border"
  id={undefined}
  style={undefined}
>
  <div
    className="divider"
    id={undefined}
    style={undefined}
  >
    <div
      className="groupTitle"
    >
      Dinosaurs
    </div>
    <div
      className="listItem"
      id={undefined}
      onClick={undefined}
      style={undefined}
    >
      <div
        className="inlineBlock"
      >
        <div
          className="avatar"
          title="Blue"
        >
          <span
            className="letter"
          >
            B
          </span>
        </div>
      </div>
      <div
        className="inlineBlock"
      >
        <button
          className="button link link_hover"
          href={undefined}
          id={undefined}
          name={undefined}
          onClick={undefined}
          style={undefined}
          tabIndex={0}
          type="button"
        >
          Blue
        </button>
      </div>
      <div
        className="rightItem"
      >
        <div
          className="switch"
        >
          <div
            className="sliderMedium sliderEnabledOffColor"
            onClick={[Function]}
            style={undefined}
          >
            <input
              checked={false}
              className="inputMedium"
              disabled={false}
              hidden={false}
              id={undefined}
              name={undefined}
              type="checkbox"
            />
            <div
              className="handleMedium handleEnabledColor"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <div
              className="onColorMedium"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="listItem"
      id={undefined}
      onClick={undefined}
      style={undefined}
    >
      <div
        className="inlineBlock"
      >
        <div
          className="avatar"
          title="Earl"
        >
          <span
            className="letter"
          >
            E
          </span>
        </div>
      </div>
      <div
        className="inlineBlock"
      >
        <button
          className="button link link_hover"
          href={undefined}
          id={undefined}
          name={undefined}
          onClick={undefined}
          style={undefined}
          tabIndex={0}
          type="button"
        >
          Earl Sinclair
        </button>
      </div>
      <div
        className="rightItem"
      >
        <div
          className="switch"
        >
          <div
            className="sliderMedium sliderEnabledOffColor"
            onClick={[Function]}
            style={undefined}
          >
            <input
              checked={false}
              className="inputMedium"
              disabled={false}
              hidden={false}
              id={undefined}
              name={undefined}
              type="checkbox"
            />
            <div
              className="handleMedium handleEnabledColor"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <div
              className="onColorMedium"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
